src/tools/tidy/src/deps.rs

```diff
@@ -600,6 +606,69 @@ pub fn check(root: &Path, cargo: &Path, bad: &mut bool) {
     assert!(checked_runtime_licenses);
 }
 
+/// Ensure the list of proc-macro crate transitive dependencies is up to date
+fn check_proc_macro_dep_list(root: &Path, cargo: &Path, bless: bool, bad: &mut bool) {
+    let mut cmd = cargo_metadata::MetadataCommand::new();
+    cmd.cargo_path(cargo)
+        .manifest_path(root.join("Cargo.toml"))
+        .features(cargo_metadata::CargoOpt::AllFeatures)
+        .other_options(vec!["--locked".to_owned()]);
+    let metadata = t!(cmd.exec());
+    let is_proc_macro_pkg =
+        |pkg: &&Package| pkg.dependencies.iter().any(|dep| dep.name.as_str() == "proc-macro2");
```

Comment:
Maybe check `pkg.targets.iter().any(|target| target.kind.iter().any(|kind| kind == TargetKind::ProcMacro))`? Not ever proc-macro necessarily depends on the proc-macro2 crate and not every crate that depends on the proc-macro2 crate is necessarily a proc-macro.

Reply:
Ah this wasn't possible in the version of cargo_metadata we were using 😄 Updated to get this; good spot